package.json

```diff
@@ -39,6 +40,7 @@
     "@vue/compiler-sfc": "^3.0.5",
     "@vue/test-utils": "^2.0.0-beta.14",
     "babel-jest": "^26.6.3",
+    "bootstrap": "git+https://github.com/twbs/bootstrap.git",
```

Comment:
Hi @ryneeverett thanks for the PR.

Is this the primary reason for this PR? The ability to bring in a package.json dependency from a git URL?

We originally wanted to bring node2nix into the nix expressions (https://github.com/MatrixAI/TypeScript-Demo-Lib/issues/5) because this makes it easier to package this all up in our nix overlay.

Need to consider the pros & cons here of this change.

Reply:
Still I wonder if this can be fixed in upstream node2nix and allow us to continue using node2nix in the same way.

Reply:
> Is this the primary reason for this PR? The ability to bring in a package.json dependency from a git URL?

Yes. There may be other reasons but I'm not aware of them. I'm working from the assumption that this project aims to be universally generalizable but that may not be the case.

> Still I wonder if this can be fixed in upstream node2nix and allow us to continue using node2nix in the same way.

Perhaps. The initial problem I see is that git dependencies in package-lock.json don't get an "integrity" field, so maybe add a `--no-network` option to node2nix so it doesn't attempt to fetch the hash (which I believe is not actually required for fetchgit).

Reply:
> Perhaps. The initial problem I see is that git dependencies in package-lock.json don't get an "integrity" field, so maybe add a --no-network option to node2nix so it doesn't attempt to fetch the hash (which I believe is not actually required for fetchgit).

How would `node2nix` fetch the data? Trust on first use? Or trust whatever npm provides?

> Yes. There may be other reasons but I'm not aware of them. I'm working from the assumption that this project aims to be universally generalizable but that may not be the case.

Mostly it's used by our projects here at Matrix AI. We have have a number of other templates that we need to maintain similar behaviour and standards, so accepting this PR may happen if we hit a dependency we need from git. You are welcome to maintain a fork in the mean time. But I reckon if this can be fixed upstream, there would no problems!

Reply:
> > Perhaps. The initial problem I see is that git dependencies in package-lock.json don't get an "integrity" field, so maybe add a --no-network option to node2nix so it doesn't attempt to fetch the hash (which I believe is not actually required for fetchgit).
> 
> How would `node2nix` fetch the data? Trust on first use? Or trust whatever npm provides?

I happened to stumble upon [this PR](https://github.com/NixOS/nixpkgs/pull/119522) which appears to attempt a similar approach with yarn. It doesn't address the problem of git dependencies but my idea of a cheat for those would be to use fetchgit without a hashsum and thus eliminate node2nix's need to download the source. I was just spitballing and I don't think I'm interested in pursuing this at the moment.

> We have have a number of other templates that we need to maintain similar behaviour and standards, so accepting this PR may happen if we hit a dependency we need from git.

AFAIK you'd run into the same issue if any recursive dependency had a git dependency so this problem may be hard to avoid.

Reply:
I just discovered https://github.com/tweag/npmlock2nix which appears to address the problem of git dependencies.

Closing for now as I don't think this PR is going anywhere at the moment.

Reply:
That looks useful. Might try it out later in the future @tegefaulkes https://github.com/tweag/npmlock2nix